### PR TITLE
Feature/dump model dot format support

### PR DESCRIPTION
**Pull request by Nityahapani**
# Fix dump_model corrupting output for dump_format='dot'

## Summary
`get_dump()` explicitly documents three supported formats: `'text'`,
`'json'`, or `'dot'`. `dump_model()`'s docstring only documented two
(`'text'` or `'json'`), and its file-writing logic only had special-cased
handling for `'json'` — anything else, including `'dot'`, fell into the
`'text'`-style branch, which prepends a `"booster[i]:\n"` header before
each tree's dump.

That header is **not valid DOT syntax**. Each tree's DOT dump is already a
complete, self-contained `digraph {...}` block (confirmed by reading the
actual C++ dump template in `src/tree/tree_model.cc`), so writing
`dump_model(fout, dump_format='dot')` silently produced a corrupted file
that graphviz/dot tooling couldn't parse, with no error or warning.

## Change
- For a **single-tree** booster, write the DOT content directly with no
  extra header — this is the common case and now produces a valid,
  directly-usable `.dot` file.
- For a **multi-tree** booster, raise a clear `ValueError` explaining that
  concatenating multiple `digraph {}` blocks into one file isn't valid DOT,
  and pointing to `get_dump()`/`to_graphviz()`/`plot_tree()` with
  `tree_idx` as the way to get an individual tree's dump instead of
  silently writing something broken.
- Updated the docstring to mention `'dot'` as a supported format, with a
  `versionadded` note.

## Commits
1. **Fix `dump_model` corrupting output for `dump_format='dot'`** — the fix
   itself, described above.
2. **Add tests** — `dump_model()` had **zero** test coverage anywhere in
   the test suite before this (double-checked: grepped for calls to the
   method specifically, not just the substring `"dump_model"`, which also
   matches an unrelated local variable name in `test_with_dask.py`). Added
   `test_dump_model_dot_format` covering both the single-tree success case
   (asserting file content matches `get_dump()` exactly) and the
   multi-tree error case (asserting a clear error and nothing corrupted
   left on disk).

## Verification

- Single-tree dot: writes clean, unmodified DOT content
- Multi-tree dot: raises `ValueError` with nothing written to the output
  buffer (confirmed via a `StringIO` stand-in)
- Text and json formats: byte-for-byte unaffected by this change

`mypy` and `ruff` both pass on all modified files.

### CODE REVIEW

File: python-package/xgboost/core.py [hunk 1]
```diff
@@ -2974,6 +2960,25 @@ def dump_model(
...
                     fout_obj.write(",\n")
             fout_obj.write("\n]")
+        elif dump_format == "dot":
+            # Each tree's dump is already a complete, self-contained `digraph {...}`
+            # block (see `to_graphviz`/`plot_tree` for the single-tree case). Unlike
+            # `text`/`json`, these can't be concatenated into one valid multi-tree
+            # DOT file with a "booster[i]:" style header, since that isn't valid DOT
+            # syntax and would corrupt the output. Support the common single-tree
+            # case directly; for multiple trees, point the user at the per-tree API
+            # instead of silently writing a broken file.
```
Comment (trivialfis) [inline python-package/xgboost/core.py]: Please don't make excessive comments.

File: tests/python/test_basic_models.py [hunk 2]
```diff
@@ -269,6 +269,37 @@ def validate_json(obj: dict) -> None:
...
+        assert "booster[" not in content
+
+        # A multi-tree booster: dump_format="dot" can't represent more than one
+        # tree in a single valid DOT file, so this should raise a clear error
+        # instead of silently writing a broken one.
+        multi_tree = xgb.train(
+            {"objective": "reg:squarederror"}, Xy, num_boost_round=3
+        )
+        out_path_multi = tmp_path / "multi_tree.dot"
+        with pytest.raises(ValueError, match="only supports writing a single tree"):
```
Comment (trivialfis) [inline tests/python/test_basic_models.py]: A single 2-3 line raise test case in `test_plotting.py` is sufficient.

**Review by trivialfis — CHANGES_REQUESTED** [review]

**Comment by Nityahapani** [comment]
Done — trimmed the comment to 2 lines, and moved the multi-tree error check to a small standalone test in test_plotting.py (test_dump_model_dot_format_multi_tree), leaving test_dump_model_dot_format focused on just the single-tree success case in test_basic_models.py.
